src/hardware/adc.rs

```diff
@@ -29,44 +51,66 @@ pub enum InputChannel {
     Seven = 7,
 }
 
+// Physical ADC devices on Thermostat
+#[derive(Clone, Copy)]
+
+pub enum AdcPhy {
+    Zero = 0,
+    One = 1,
+    Two = 2,
+    Three = 3,
+}
+
 type O = Output<PushPull>;
-type Adcs = (
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE0<O>>,
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE1<O>>,
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE3<O>>,
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE4<O>>,
-);
+type Adcs = ad7172::Ad7172<Spi<SPI4, Enabled>>;
 type SpiPins = (
     PE2<Alternate<AF5>>,
     PE5<Alternate<AF5>>,
     PE6<Alternate<AF5>>,
 );
-
+/// All pins for all ADCs.
+/// * `spi` - Spi clk, miso, mosi (in this order).
+/// * `cs` - The four chip select pins.
+/// * `rdyn` - ADC rdyn input (this is the same ad spi dout).
+/// * `sync` - ADC sync pin (shared for all adc phys).
 pub struct AdcPins {
+    pub spi: SpiPins,
     pub cs: (PE0<O>, PE1<O>, PE3<O>, PE4<O>),
+    pub rdyn: PC11<Input<PullUp>>,
+    pub sync: PB11<O>,
 }
 
 pub struct Adc {
     pub adcs: Adcs,
+    pub rdyn: PC11<Input<PullUp>>,
+    pub sync: PB11<O>,
+    pub cs: (PE0<O>, PE1<O>, PE3<O>, PE4<O>),
+    pub current_position: usize, // Schedule position
 }
 
 impl Adc {
+    const SCHEDULE: [(AdcPhy, InputChannel); 8] = [
```

Comment:
This needs documentation! Much more than the out-of-date docstrings elsewhere.

Reply:
I wrote some minimal documentation about the schedule and the interrupt sampling as they depend on each other. Since things might still change a lot I would write a proper documentation at a later point.